src/main/java/org/dave/compactmachines3/block/BlockWall.java

```diff
@@ -104,8 +104,9 @@ public boolean onBlockActivated(World world, BlockPos pos, IBlockState state, En
                 playerStack.setCount(playerStack.getCount()-1);
                 WorldSavedDataMachines.getInstance().addTunnel(pos, tunnelSide);
             } else {
-                // TODO: Localization
-                player.sendStatusMessage(new TextComponentString(TextFormatting.RED + "All tunnels already used up"), false);
+                TextComponentTranslation tc = new TextComponentTranslation("hint.compactmachines3.all_tunnels_used");
+                tc.getStyle().setColor(TextFormatting.RED);
```

Comment:
Should probably prefer `mergeFormatting` here, since it's a bit shorter and will keep defaults as MC changes

Reply:
Unfortunately, this doesn't exist for 1.12 as far as I know and checked.